types/defines/vectorize.d.ts

```diff
@@ -177,3 +196,50 @@ declare abstract class VectorizeIndex {
    */
   public getByIds(ids: string[]): Promise<VectorizeVector[]>;
 }
+
+/**
+ * A Vectorize Vector Search Index for querying vectors/embeddings.
+ *
+ * Mutations in this version are async, returning a mutation id.
+ */
+declare abstract class Vectorize {
```

Comment:
There's a lot of duplication here—did the generics and aliasing approach not work out for some reason?

Reply:
The generics approach would work fine - however, we would lose out on the specificity (and correctness) of some of the tsdoc comments if we took that approach. Compare https://github.com/cloudflare/workerd/pull/2421/files#diff-d770090d5d2a9db9ac5822eb39c8f82411484c0dab8a12cbfd618e924eda58a5R232 & https://github.com/cloudflare/workerd/pull/2421/files#diff-4386ee8e0b9580e2b88770e50a5e19d669bd8d8957df5e215b633330a9f2cd23R224